node/src/main/scala/org/bitcoins/node/networking/P2PClient.scala

```diff
@@ -88,13 +95,21 @@ case class P2PClientActor(
         val networkMsg = NetworkMessage(network, payload)
         self.forward(networkMsg)
       case message: Tcp.Message =>
-        val newUnalignedBytes =
+        val newUnalignedBytesOpt =
           handleTcpMessage(message, peerConnection, unalignedBytes)
-        context.become(awaitNetworkRequest(peerConnection, newUnalignedBytes))
+        newUnalignedBytesOpt match {
+          case Some(newUnalignedBytes) =>
+            context.become(
+              awaitNetworkRequest(peerConnection, newUnalignedBytes))
+          case None =>
+            //signifies the tcp connection was closed, go back to our initial state
+            context.become(receive)
+        }
+
       case metaMsg: P2PClient.MetaMsg =>
         sender() ! handleMetaMsg(metaMsg)
       case Terminated(actor) if actor == peerConnection =>
-        context stop self
+        context.stop(self)
```

Comment:
Should we stop here? Probably we shoud reconnect if the peer connection was closed.

Reply:
Something like
```scala
case Terminated(actor) if actor == peerConnection =>
  // clear all mutable state here
  self() ! P2PClient.ConnectCommand 
  context.become(receive)
```